src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -66,34 +76,66 @@ internal static class CodeActionHelpers
 
         private static VSCodeAction GenerateVSCodeAction(
             CodeActionParams request,
-            CodeAction codeAction,
-            CodeActionKind codeActionKind,
+            SourceText documentText,
+            IUnifiedSuggestedAction suggestedAction,
+            LSP.CodeActionKind codeActionKind,
+            LSP.Range? applicableRange,
+            int currentSetNumber,
+            ref int highestSetNumber,
             string currentTitle = "")
         {
-            using var _ = ArrayBuilder<VSCodeAction>.GetInstance(out var nestedActions);
-
             if (!string.IsNullOrEmpty(currentTitle))
             {
                 // Adding a delimiter for nested code actions, e.g. 'Suppress or Configure issues|Suppress IDEXXXX|in Source'
                 currentTitle += '|';
             }
 
+            var codeAction = suggestedAction.OriginalCodeAction;
             currentTitle += codeAction.Title;
 
             // Nested code actions' unique identifiers consist of: parent code action unique identifier + '|' + title of code action
-            foreach (var action in codeAction.NestedCodeActions)
-            {
-                nestedActions.Add(GenerateVSCodeAction(request, action, codeActionKind, currentTitle));
-            }
+            var nestedActions = GenerateNestedVSCodeActions(request, documentText, suggestedAction, codeActionKind, ref highestSetNumber, currentTitle);
 
             return new VSCodeAction
             {
                 Title = codeAction.Title,
                 Kind = codeActionKind,
                 Diagnostics = request.Context.Diagnostics,
-                Children = nestedActions.ToArray(),
+                Children = nestedActions,
+                Priority = CodeActionPriorityToPriorityLevel(codeAction.Priority),
+                Group = "Roslyn" + currentSetNumber.ToString(),
```

Comment:
Currently grouping is done by prefixing "Roslyn" with the group number. "Roslyn" was prefixed just to make sure no other language servers use the same group name. If anyone has better suggestions for group names, please let me know.

Reply:
this seems reasonable.